### PR TITLE
Fix - Optimized NetworkTransform on all networked prefabs in the Invaders sample [MTT-7599] 

### PR DESCRIPTION
**Pull request by bastienunity**
### Description
The gameplay feeling is very different on clients vs. host in the Invaders Sample.
This PR is changing the interpolation of netwrok transforms to false in order to fix the issue.
Because the game is not using any physics or animations to move transforms in the scene, the server itself is not doing any interpolation between positions and simply forces the new transform position of enemies or bullets. Removing the interpolation prevents the client from smoothing that movement locally and makes all movements looks like they are behaving the same on the host and any client.

While updating the NetworkTransform, we also removed most of the sync values of the transform to keep only the ones that changes during the gameplay.
Players and SuperEnemy only need to sync their postion.x because they can't move vertically.
Enemies only need to sync their position.x and position.y.
Shields don't move, their NetworkTransform has been removed.
Bullets only need to sync their position.y because they only move vertically.

Before changes Host vs. Client

https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/assets/29260853/06d6d197-5afa-4b9f-a49d-b2c1892fa0f3

After these changes Host vs. Client

https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/assets/29260853/c25f51d8-1cc9-4d54-aff7-4e39901f6f40

### Issue Number(s)
[MTT-7599](https://jira.unity3d.com/browse/MTT-7599) - Turned interpolation to false on all the networked transform to make movements between host and client render movements the same way.

### Contribution checklist
 - [ ] Tests have been added for the project and/or any internal package
 - [x] Release notes have been added to the [project changelog](../CHANGELOG.md) file
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] JIRA ticket ID is in the PR title or at least one commit message
 - [x] Include the ticket ID number within the body message of the PR to create a hyperlink

### CODE REVIEW

**Review by Elfi0Kuhndorf — APPROVED** [review]
Looks smooth!

**Review by fernando-cortez — DISMISSED** [review]
I'm sure that they were improvements, but what were the changes to Invaders Game Manager and SceneTransitionHandler prefabs?
Just listed would be fine.

**Comment by bastienunity** [comment]
@fernando-cortez I think they got reserialized because I touched them, but haven't changed anything, the diff is only the component order that changed, not sure why.
I can revert them to avoid the noise in the PR.

**Comment by fernando-cortez** [comment]
> @fernando-cortez I think they got reserialized because I touched them, but haven't changed anything, the diff is only the component order that changed, not sure why.
> I can revert them to avoid the noise in the PR.

Ah that makes sense! I think that's fine if they go through.

File: Basic/Invaders/Assets/Prefabs/SceneTransitionHandler.prefab [hunk 1]
```diff
@@ -9,8 +9,8 @@ GameObject:
   serializedVersion: 6
   m_Component:
   - component: {fileID: 7334280962834855337}
-  - component: {fileID: 7334280962834855336}
   - component: {fileID: 7334280962834855351}
+  - component: {fileID: 7334280962834855336}
```
Comment (RikuTheFuffs) [inline Basic/Invaders/Assets/Prefabs/SceneTransitionHandler.prefab:13]: it seems that the order of components changed here. Is there a reason of that? Does this solve anything? Otherwise it would be better to revert this change, as changign the order of components on a gameObjects alters the execution order and can lead to bugs.

Comment (bastienunity) [reply Basic/Invaders/Assets/Prefabs/SceneTransitionHandler.prefab]: Agree, not sure why it changed, Unity just decided to reserialize them at some point. I've reverted that change and tested the project again, which seems fine.

File: Basic/Invaders/Assets/Prefabs/bullet.prefab [hunk 2]
```diff
@@ -10,10 +10,10 @@ GameObject:
   m_Component:
   - component: {fileID: 400000}
   - component: {fileID: 21200000}
+  - component: {fileID: 2859882261451710602}
```
Comment (RikuTheFuffs) [inline Basic/Invaders/Assets/Prefabs/bullet.prefab:13]: @bastienunity  it seems the order of scripts was changed here as well, is this intended? If not, can you revert it to avoid potential issues?

File: Basic/Invaders/Assets/Prefabs/enemy1.prefab [hunk 3]
```diff
@@ -10,11 +10,11 @@ GameObject:
   m_Component:
   - component: {fileID: 400000}
   - component: {fileID: 21200000}
+  - component: {fileID: 2037077820506562951}
```
Comment (RikuTheFuffs) [inline Basic/Invaders/Assets/Prefabs/enemy1.prefab:13]: same as above

File: Basic/Invaders/Assets/Prefabs/enemy2.prefab [hunk 4]
```diff
@@ -10,11 +10,11 @@ GameObject:
   m_Component:
   - component: {fileID: 400000}
   - component: {fileID: 21200000}
+  - component: {fileID: 1361474663871475650}
```
Comment (RikuTheFuffs) [inline Basic/Invaders/Assets/Prefabs/enemy2.prefab:13]: same as above

File: Basic/Invaders/Assets/Prefabs/enemy3.prefab [hunk 5]
```diff
@@ -10,11 +10,11 @@ GameObject:
   m_Component:
   - component: {fileID: 400000}
   - component: {fileID: 21200000}
+  - component: {fileID: 7651551376209856180}
```
Comment (RikuTheFuffs) [inline Basic/Invaders/Assets/Prefabs/enemy3.prefab:13]: same as above

File: Basic/Invaders/Assets/Prefabs/enemyBullet.prefab [hunk 6]
```diff
@@ -11,9 +11,9 @@ GameObject:
   - component: {fileID: 400000}
   - component: {fileID: 21200000}
   - component: {fileID: 6100000}
+  - component: {fileID: 2336033219118315817}
```
Comment (RikuTheFuffs) [inline Basic/Invaders/Assets/Prefabs/enemyBullet.prefab:14]: same as above

File: Basic/Invaders/Assets/Prefabs/player.prefab [hunk 7]
```diff
@@ -10,10 +10,10 @@ GameObject:
   m_Component:
   - component: {fileID: 400000}
   - component: {fileID: 21200000}
+  - component: {fileID: 5053919295125167455}
```
Comment (RikuTheFuffs) [inline Basic/Invaders/Assets/Prefabs/player.prefab:13]: same as above

File: Basic/Invaders/Assets/Prefabs/shield.prefab [hunk 8]
```diff
@@ -14,7 +14,6 @@ GameObject:
   - component: {fileID: 6100000}
   - component: {fileID: 11400000}
   - component: {fileID: 6564576217164106930}
-  - component: {fileID: 8844355272195160781}
```
Comment (RikuTheFuffs) [inline Basic/Invaders/Assets/Prefabs/shield.prefab:17]: what component was removed here?

File: Basic/Invaders/Assets/Prefabs/superEnemy.prefab [hunk 9]
```diff
@@ -11,11 +11,11 @@ GameObject:
   - component: {fileID: 400000}
   - component: {fileID: 21200000}
   - component: {fileID: 5000000}
+  - component: {fileID: 3955316467729786863}
```
Comment (RikuTheFuffs) [inline Basic/Invaders/Assets/Prefabs/superEnemy.prefab:14]: same as above

**Review by RikuTheFuffs — CHANGES_REQUESTED** [review]

File: Basic/Invaders/Assets/Prefabs/bullet.prefab (hunk 2 above)
Comment (bastienunity) [reply Basic/Invaders/Assets/Prefabs/bullet.prefab:13]: So, funny story... When selecting the prefab, the Inspector refreshes after one frame and moves the NetworkObject component in third place on every single Prefab I select in the project. So any further changes made to the prefab will save the NetworkObject component position changed...
I can manually revert that change outside of Unity, but because it is forced and automatic, I think it's better to just commit it?

File: Basic/Invaders/Assets/Prefabs/shield.prefab (hunk 8 above)
Comment (bastienunity) [reply Basic/Invaders/Assets/Prefabs/shield.prefab:17]: I removed the NetworkTransform, Shields are spawned and destroyed but never move, so they don't need to sync their Transform at all.

File: Basic/Invaders/Assets/Prefabs/bullet.prefab (hunk 2 above)
Comment (fernando-cortez) [reply Basic/Invaders/Assets/Prefabs/bullet.prefab:13]: If Netcode is forcing reserialization, and we're going to keep running into this change in subsequent PRs, this can go through.
Let's expose our code to potential race conditions and address them, if they arise!

Comment (fernando-cortez) [reply Basic/Invaders/Assets/Prefabs/bullet.prefab:13]: @RikuTheFuffs

Comment (RikuTheFuffs) [reply Basic/Invaders/Assets/Prefabs/bullet.prefab:13]: yeah, if it's automatic there's little we can do. I just wanted to be sure that it was not a manual change.

**Review by RikuTheFuffs — APPROVED** [review]
